docs/autoprocess.py

```diff
@@ -0,0 +1,397 @@
+"""
+Sphinx extension for semi-autmoatic documentation of resolwe-bio processes.
+
+This module introduces new directive that enables user to document a
+set of processes. The directive can be used in three variations::
+
+    .. autoprocess::
+
+    .. autoprocess:: category_name
+
+    .. autoprocess:: process_slug
+
+The first option documents *all* processes.
+
+The second one documents just the ones that have *category* equal to
+`category_name`. Since some categories have subcategories (analyses,
+analyses:alignment, analyses:variants), this means that subcategories
+will *not* be documented.
+
+Third option decuments only one process: the one with *slug* equal to
+`process_slug`.
+"""
+
+import os
+import uuid
+import fnmatch
+
+import six
+import yaml
+
+from docutils import nodes
+from docutils.parsers.rst import Directive
+
+
+PROCESS_DIR = os.path.normpath(os.path.dirname(__file__) + "./../resolwe_bio/processes/")
+SOURCE_BASE_URL = 'https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/'
```

Comment:
Is there an option to pass this values as arguments in `rst` file? This directive will be also used in other packages that contains processes, so `resolwe-bio` shouldn't be hardcoded.